packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWRelatedCommunityCard/CWRelatedCommunityCard.tsx

```diff
@@ -7,106 +8,162 @@ import {
   MixpanelClickthroughEvent,
   MixpanelClickthroughPayload,
 } from '../../../../../../../shared/analytics/types';
+import { useCommunityStake } from '../../../CommunityStake';
+import useJoinCommunity from '../../../SublayoutHeader/useJoinCommunity';
 import { CWCommunityAvatar } from '../../cw_community_avatar';
 import { CWIcon } from '../../cw_icons/cw_icon';
 import { CWText } from '../../cw_text';
 import { ComponentType } from '../../types';
+import { CWButton } from '../cw_button';
 import './CWRelatedCommunityCard.scss';
 import { addPeriodToText } from './utils';
 
 type CWRelatedCommunityCardProps = {
-  id: string;
-  communityName: string;
-  communityIconUrl: string;
-  communityDescription: string;
+  community: ChainInfo | CommunityData;
   memberCount: string;
+  connected?: boolean;
   threadCount: string;
-  actions?: JSX.Element;
+  stakeValue?: number;
+  stakeChange?: number;
 };
 
 export const CWRelatedCommunityCard = ({
-  id,
-  communityName,
-  communityIconUrl,
-  communityDescription,
+  community,
+  connected,
   memberCount,
   threadCount,
-  actions,
+  stakeValue,
+  stakeChange,
 }: CWRelatedCommunityCardProps) => {
   const navigate = useCommonNavigate();
-  const communityAvatar = {
-    iconUrl: communityIconUrl,
-    name: communityName,
-  } as ChainInfo;
+  const { stakeEnabled } = useCommunityStake({ communityId: community.id });
 
   const { trackAnalytics } =
     useBrowserAnalyticsTrack<MixpanelClickthroughPayload>({
       onAction: true,
     });
 
   const handleClick = useCallback(
-    (e, communityId: string) => {
+    (e) => {
       e.preventDefault();
       trackAnalytics({
         event: MixpanelClickthroughEvent.DIRECTORY_TO_COMMUNITY_PAGE,
       });
       if (isCommandClick(e)) {
-        window.open(`/${communityId}`, '_blank');
+        window.open(`/${community.id}`, '_blank');
```

Comment:
Can't we use the `useCommonNavigate` hook here?

Reply:
Unfortunately, I don't think we can.